app/models/translation.rb

```diff
@@ -54,6 +55,26 @@ def validate_translated_is_translatable
     end
   end
 
+  def validate_string_versions
+    unless string_versions.is_a?(Hash)
+      errors.add(:string_versions, "must be present but can be empty")
+      return
```

Comment:
is this return needed? If it's a hash it'll return nil as well via implicit return in ruby.

Reply:
If it's not a hash the rest of the validation would raise an exception when trying to use it as a hash.